app/views/contexts/preview_context.py

```diff
@@ -0,0 +1,60 @@
+from typing import Generator, Mapping, Optional, Union
+
+from app.data_models import AnswerStore, ListStore, ProgressStore
+from app.data_models.metadata_proxy import MetadataProxy
+from app.questionnaire import QuestionnaireSchema
+from app.views.contexts import Context
+from app.views.contexts.section_preview_context import SectionPreviewContext
+
+
+class PreviewNotEnabledException(Exception):
+    pass
+
+
+class PreviewContext(Context):
+    def __init__(
+        self,
+        language: str,
+        schema: QuestionnaireSchema,
+        answer_store: AnswerStore,
+        list_store: ListStore,
+        progress_store: ProgressStore,
+        metadata: Optional[MetadataProxy],
+        response_metadata: Mapping[str, Union[str, int, list]],
+    ):
+        if not schema.preview_enabled:
+            raise PreviewNotEnabledException(404)
```

Comment:
This exception needs to be handled in errors.py. Also the `404` here doesn't do or mean much, should be removed.

Reply:
Added to errors.py plus new integration test asserting 500, was I right to use `404` template for it?